### PR TITLE
[cli] Add `Output#link()` function to format terminal hyperlinks

### PR DESCRIPTION
**Pull request by TooTallNate**
This adds a `link()` helper function to the `Output` class that is inspired by the `terminal-link` npm package.

The main difference with this version is that it's more tightly integrated with the `Output` class for the purposes of being able to toggle hyperlinks support on/off for [unit tests](https://github.com/vercel/vercel/blob/4a54b19f46953af2a808bad3e461798e5a40e350/packages/cli/test/unit/util/output/create-output.test.ts) by setting the `output.supportsHyperlink` boolean.

> **Note:** Since hyperlinks are still a relatively new feature, and users might not yet understand how to interact with them, we should only use this function for progressive enhancement scenarios at this time, and _not_ as part of a critical UX.

### CODE REVIEW

File: packages/cli/src/util/output/create-output.ts [hunk 1]
```diff
@@ -164,4 +184,28 @@ export class Output {
...
+
+  /**
+   * Returns an ANSI formatted hyperlink when support has been enabled.
+   */
+  link = (
+    text: string,
+    url: string,
+    { fallback }: LinkOptions = {}
+  ): string => {
+    // Credit: https://github.com/sindresorhus/terminal-link
```
Comment (MatthewStanciu) [inline packages/cli/src/util/output/create-output.ts]: I think we should start following a convention that's something like [how @styfle and I attributed part of the `vc env pull` code](https://github.com/vercel/vercel/pull/8170/files#diff-32c1018d233f2c92e3dd39a35fa9abe29eeb1e9dc26d49e603b22023e860fb83). Something like

```
Originally authored by Sindre Sorhus under [license]: [link to license]
[link to piece of code]
```

**Review by MatthewStanciu — APPROVED** [review]

File: packages/cli/src/util/output/create-output.ts (hunk 1 above)
Comment (EndangeredMassa) [reply packages/cli/src/util/output/create-output.ts]: This doesn't seem like a credit-worth amount or significance of code to me, but I'm happy to include it if anyone would like to. If we keep it, an expanded comment (as Matthew mentioned) would probably be best.

**Review by EndangeredMassa — APPROVED** [review]
Are we going to use this? If the plan is "not for now", I'd rather close this until we need it.

**Comment by TooTallNate** [comment]
I think we should be.

**Review by javivelasco — APPROVED** [review]

**Comment by TooTallNate** [comment]
Btw [`xterm.js` has a PR to support hyperlinks](https://github.com/xtermjs/xterm.js/pull/4005), so they will be supported in VSCode's terminal soon.

**Comment by MatthewStanciu** [comment]
I expect the answer to "Are we going to use this?" to be "Not for now", but even if that's the case I think this should be merged and we should start using terminal links where they can be used.

Related: a good hackathon project or part of a hackathon project could be making the CLI sparkle. Examples could include adding little details like Turbo's "FULL TURBO" thing, and adding terminal links.

File: packages/cli/src/util/output/create-output.ts [hunk 2]
```diff
@@ -49,15 +61,19 @@ export class Output {
     str: string,
     slug: string | null = null,
     link: string | null = null,
-    action: string | null = 'Learn More'
+    action: string = 'Learn More'
```
Comment (styfle) [inline packages/cli/src/util/output/create-output.ts]:
```suggestion
    action = 'Learn More'
```

File: packages/cli/src/util/output/create-output.ts [hunk 3]
```diff
@@ -49,15 +61,19 @@ export class Output {
...
     const details = slug ? `https://err.sh/vercel/${slug}` : link;
 
     this.print(
       chalk.yellow(
         chalk.bold('WARN! ') +
           str +
-          (details ? `\n${action}: ${renderLink(details)}` : '')
+          (details
+            ? `\n${this.link(action, details, {
+                fallback: () => `${action}: ${renderLink(details)}`,
```
Comment (styfle) [inline packages/cli/src/util/output/create-output.ts]: How will users know to click the link? I think we need to be careful to only use this when the link matches the text.

Comment (TooTallNate) [reply packages/cli/src/util/output/create-output.ts]: I reverted the change to `warn()` and `error()`. We can discuss that at a later time.

**Review by cb1kenobi — APPROVED** [review]
LGTM

File: packages/cli/src/util/output/create-output.ts [hunk 4]
```diff
@@ -196,8 +196,6 @@ export class Output {
     url: string,
     { fallback }: LinkOptions = {}
   ): string => {
-    // Credit: https://github.com/sindresorhus/terminal-link
```
Comment (styfle) [inline packages/cli/src/util/output/create-output.ts]: I think we need to leave this here if its copy pasta. Should say MIT license

**Review by styfle — APPROVED** [review]
Approving based on this important caveat:

> Note: Since hyperlinks are still a relatively new feature, and users might not yet understand how to interact with them, we should only use this function for progressive enhancement scenarios at this time, and not as part of a critical UX

We need to be careful how to use this and I think a good start would be @TooTallNate's idea for `--help` to make options/flags clickable and link to the docs.

**Pull request merged**